app/api/routes/image_queries.py

```diff
@@ -108,10 +109,12 @@ async def post_image_query(
         },
     )
 
+    edge_only = os.environ.get("EDGE_ONLY", "") == "ENABLED"
```

Comment:
FWIW, if we did want to restrict this logic to only run when want_async is requested, it would be as simple as:
```
edge_only = (os.environ.get("EDGE_ONLY", "") == "ENABLED") and _want_async
```

Reply:
Ah good point, that is easy actually. I think for now I'm fine with making all methods use the flag, but we could consider changing this in the future if we want to make it more restricted.